src/app/basket/basket.component.ts

```diff
@@ -45,4 +54,50 @@ export class BasketComponent implements OnInit, AfterViewInit {
   public getKeys(object: any) {
     return Object.keys(object);
   }
+
+  private createQuery(object: any): string {
+    let query = '';
+    for (const key of this.getKeys(object)) {
+      if (object[key] !== undefined) {
+        let queryPerObject = ' complex_id:';
+        queryPerObject += object[key].id;
+        query += queryPerObject;
+      }
+    }
+    return query;
+  }
+
+  private requestComplexesForNavigator() {
+    this.complexPortalService.findComplex(this.createQuery(this._complexBasket)).subscribe(complexSearch => {
```

Comment:
To fix the issue with only displaying 10 complexes in the basket, we need to set the the page size to the right value. By default it is 10.
```
const pageSize = Object.values(this._complexBasket).length;
    this.complexPortalService.findComplex(this.createQuery(this._complexBasket), [], [], [], 1, pageSize)
      .subscribe(complexSearch => {
```